sdk/cosmos/azure_data_cosmos/src/operation_context.rs

```diff
@@ -32,4 +36,23 @@ impl OperationType {
                 | OperationType::HeadFeed
         )
     }
+
+    /// Maps the `OperationType` to its corresponding HTTP verb.
+    pub fn http_method(&self) -> Method {
```

Comment:
nit: Since `OperationType` derives `Copy`, you can just take `self` by value here:

```suggestion
    pub fn http_method(self) -> Method {
```

When you do that, you'll still be able to write code like this:

```rust
let op = OperationType::HeadFeed;
let method = op.method();
println!("op = {}", op);
println!("method = {}", method);
```

Normally, this would cause a compiler error if `.method()` took `self` by-value, because Rust would **move** the value and `op` would become unusable. However, because `OperationType` derives from `Copy` (because it's trivially small), Rust will actually make a copy of it when it calls `.method()`, and `op` remains usable.

Reply:
Refactored. Thanks for the background here.